frontend/src/scenes/session-recordings/playlist/sessionRecordingsPlaylistLogic.ts

```diff
@@ -1,128 +1,680 @@
+                    (filters.date_from === defaultFilters.date_from && filters.date_to === defaultFilters.date_to
+                        ? 0
+                        : 1) +
+                    (filters.console_logs?.length || 0)
+                )
+            },
+        ],
+        hasAdvancedFilters: [
+            (s) => [s.filters, (_, props) => props.personUUID],
+            (filters, personUUID) => {
+                const defaultFilters = getDefaultFilters(personUUID)
+                return addedAdvancedFilters(filters, defaultFilters)
+            },
+        ],
+
+        otherRecordings: [
+            (s) => [s.sessionRecordings, s.hideViewedRecordings, s.pinnedRecordings, s.selectedRecordingId],
+            (
+                sessionRecordings,
+                hideViewedRecordings,
+                pinnedRecordings,
+                selectedRecordingId
+            ): SessionRecordingType[] => {
+                return sessionRecordings.filter((rec) => {
+                    if (pinnedRecordings.find((pinned) => pinned.id === rec.id)) {
+                        return false
+                    }
+
+                    if (hideViewedRecordings && rec.viewed && rec.id !== selectedRecordingId) {
+                        return false
+                    }
+
+                    return true
+                })
+            },
+        ],
+
+        recordings: [
+            (s) => [s.pinnedRecordings, s.otherRecordings],
+            (pinnedRecordings, otherRecordings): SessionRecordingType[] => {
+                return [...pinnedRecordings, ...otherRecordings]
+            },
+        ],
+    }),
+
+    actionToUrl(({ props, values }) => {
+        if (!props.updateSearchParams) {
+            return {}
+        }
+        const buildURL = (
+            replace: boolean
+        ): [
+            string,
+            Params,
+            Record<string, any>,
+            {
+                replace: boolean
+            }
+        ] => {
+            const params: Params = objectClean({
+                filters: values.customFilters ?? undefined,
+                sessionRecordingId: values.selectedRecordingId ?? undefined,
+            })
+
+            // We used to have sessionRecordingId in the hash, so we keep it there for backwards compatibility
+            if (router.values.hashParams.sessionRecordingId) {
+                delete router.values.hashParams.sessionRecordingId
+            }
+
+            return [router.values.location.pathname, params, router.values.hashParams, { replace }]
+        }
+
+        return {
+            setSelectedRecordingId: () => buildURL(false),
+            setFilters: () => buildURL(true),
+            resetFilters: () => buildURL(true),
+        }
+    }),
+
+    urlToAction(({ actions, values, props }) => {
+        const urlToAction = (_: any, params: Params, hashParams: Params): void => {
```

Comment:
I think this is connected to my earlier comment on url params. Is this needed if we remove the ability to link to a specific recording in a playlist?

Reply:
It is but only for the standard Recent recordings screen